src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ResourceIndexNode.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.NativeFormat;
+using Internal.Text;
+using System;
+using System.IO;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a hash table of resources within the resource blob in the image.
+    /// </summary>
+    internal class ResourceIndexNode : ObjectNode, ISymbolNode
+    {
+        private ResourceDataNode _resourceDataNode;
+
+        public ResourceIndexNode(ResourceDataNode resourceDataNode)
+        {
+            _resourceDataNode = resourceDataNode;
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, "__embedded_resourceindex_End", true);
+        }
+
+        private ObjectAndOffsetSymbolNode _endSymbol;
+
+        public ISymbolNode EndSymbol => _endSymbol;
+
+        public override bool IsShareable => false;
+
+        public override ObjectNodeSection Section => ObjectNodeSection.ReadOnlyDataSection;
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public int Offset => 0;
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix).Append("__embedded_resourceindex");
+        }
+
+        protected override string GetName() => this.GetMangledName();
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            // This node has no relocations.
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolNode[] { this });
+
+            byte[] blob = GenerateIndexBlob(factory);
+            return new ObjectData(
+                blob,
+                Array.Empty<Relocation>(),
+                1,
+                new ISymbolNode[]
+                {
+                                this,
```

Comment:
Nit: Is this the whitespace formatting that VS generates for this? It looks like more whitespace than there should be.